obsproject/obs-studio
# plugins/obs-filters, UI: Add support for N sidechain ducking sources for the compressor filter

### Description
This patch adds in the ability to use N ducking sources in the compressor filter. 

### Motivation and Context
In broadcasting settings, there are often times when you have more than one ducking source into a compressor. A common example is multiple commentators commenting over a background audio track. Right now this requires expensive extra steps involved in the pipeline. Further, this is a much more obvious solution requiring less tinkering.

### How Has This Been Tested?
This has been tested in the following way:
1. create a compression target media source with the following params:
   i. threshold at min and release at max to make the effect obvious
2. Create 1 input source, then 2, then 3, then 4 and for each:
   i. Add it as a compression sidechain source to compression target
   ii. mute and unmute each individually, then all together
   iii. Pause and unpause each individually then all together.

The effect should be very obvious in the levels view if threshold is set at minimum and release at maximum.
For best effect, use both ducking sources and target with constant audio peaking the level of its given channel.

Tested on macOS (Apple silicon 14.5), Windows 11 (x86_64), and Ubuntu 24.04 (x86_64)

### Types of changes
New feature (non-breaking change which adds functionality)
Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:
